feature/eventdetail/src/commonMain/kotlin/com/adammcneilly/pocketleague/feature/eventdetail/EventDetailPresenter.kt

```diff
@@ -85,15 +86,28 @@ class EventDetailPresenter(
         return EventDetailScreen.State(
             event = displayModel,
             selectedStageIndex = selectedStageIndex,
-            matchesForSelectedStage = matchesForSelectedStage,
+            matchesForSelectedStageByDate = matchesForSelectedStage,
         ) { uiEvent ->
             when (uiEvent) {
                 is EventDetailScreen.Event.StageSelected -> {
                     selectedStageIndex = uiEvent.stageIndex
                 }
+
                 is EventDetailScreen.Event.MatchClicked -> {
                     onMatchClicked.invoke(uiEvent.matchId)
                 }
+
+                is EventDetailScreen.Event.SectionClicked -> {
+                    matchesForSelectedStage = matchesForSelectedStage.map { originalSection ->
+                        if (originalSection == uiEvent.section) {
```

Comment:
Instead of comparing on sections (which could have long lists), maybe compare just on section title?